iothub/service/src/ClientApiVersionHelper.cs

```diff
@@ -12,7 +12,7 @@ internal class ClientApiVersionHelper
     {
         private const string ApiVersionQueryPrefix = "api-version=";
         private const string ApiVersionGA = "2016-02-03";
-        private const string ApiVersionDefault = "2019-03-30";
```

Comment:
This is a different API version we were targeting from what is specified in ClientApiVersionHelper on the device side ... how does that happen?

Reply:
does that mean the device and the service clients don't necessarily target the same API versions? (even though they will after this PR)

Reply:
It means that, at some point, the service SDK needed to bump the API version to get a new feature, but the device side didn't care about that feature, or the feature didn't impact the device side